x/gamm/types/params.go

```diff
@@ -0,0 +1,70 @@
+package types
+
+import (
+	"fmt"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	paramtypes "github.com/cosmos/cosmos-sdk/x/params/types"
+	appparams "github.com/osmosis-labs/osmosis/app/params"
+)
+
+// Parameter store keys
+var (
+	KeyPoolCreationFee = []byte("PoolCreationFee")
+)
+
+// ParamTable for gamm module.
+func ParamKeyTable() paramtypes.KeyTable {
+	return paramtypes.NewKeyTable().RegisterParamSet(&Params{})
+}
+
+func NewParams(poolCreationFee sdk.Coins) Params {
+	return Params{
+		PoolCreationFee: poolCreationFee,
+	}
+}
+
+// default gamm module parameters
+func DefaultParams() Params {
+	return Params{
+		PoolCreationFee: sdk.Coins{sdk.NewInt64Coin(appparams.BaseCoinUnit, 1000_000_000)}, // 1000 OSMO
+	}
+}
+
+// validate params
+func (p Params) Validate() error {
+	if err := validatePoolCreationFee(p.PoolCreationFee); err != nil {
+		return err
+	}
+
+	return nil
+
+}
+
+// Implements params.ParamSet
+func (p *Params) ParamSetPairs() paramtypes.ParamSetPairs {
+	return paramtypes.ParamSetPairs{
+		paramtypes.NewParamSetPair(KeyPoolCreationFee, &p.PoolCreationFee, validatePoolCreationFee),
+	}
+}
+
+func validatePoolCreationFee(i interface{}) error {
```

Comment:
Inspired the deposit bug, we should also add checking that the new balance is a balance that actually exists on chain. We can do that later though.

Reply:
This is basic validation - and it's not using the kvstore - for additional check can set the validation function on SetParams probably.